ecosystem/sep-0039.md

```diff
@@ -0,0 +1,196 @@
+
+Our exclusive focus in this SEP is Step 2: **storing the asset within the account.**
+
+### Data Format
+We begin by describing the raw data format, then dive into each component:
+
+  - index (2 bytes)
+  - version (1 byte)
+  - metadata length, `n` (varying)
+  - metadata (`n` bytes)
+  - binary data (varying)
+
+For example,
+
+    00|1|26|text/plain;l=13,image/webp|TPwJh>}A"=r@@Y?Fetc...
+
+(Here, the `|` characters are semantic, included just to separate the pieces in the list above; they are not actually in the data itself.)
+
+The following sections break down each component in detail.
+
+#### Index
+Since an account can only have 1000 entries, we only need 10 bits for an index field. We _only_ need indices present so that the data entries are ASCII-sortable in Horizon. In fact, decoders can ignore it entirely if they wish. Thus, the index is expressed as a "base 36" value: 0-9 then a-z.
+
+A valid SEP-39 entry always begins with `00`, indicating that this is the 0th row of data. In fact, every row's key _must_ begin with a two-character, [Base36] index.
+
+---------
+
+**Author's note**: This portion is definitely up for debate. Base36 (well, more accurately, a radix of 36) allows us to shave a byte off of each row (saving up to 1kB total), but we could stick to a 000-999 model (stringified integer) for simplicity. We could also do `aa` through `zz`, because as noted, the indices are _purely_ for sorting. 
+
+Note that supporting a 36-radix is trivial in any language, since we know the value is strictly an integer. For example,
+
+```python
+import string 
+ALPHABET = string.digits + string.ascii_lowercase
+
+def int_to_b36(i: int) -> str:
+    d, r = divmod(i, len(ALPHABET))
+    return ALPHABET[d] + ALPHABET[r]
+
+def b36_to_int(s: str) -> int:
+    return ALPHABET.index(s[0]) * len(ALPHABET) + ALPHABET.index(s[1])
+```
+
+---------
+
+#### Version
+The first entry's index is followed by the version string. For the current version of this SEP, this is just `1`.
+
+#### Metadata
+The **metadata** is a comma-separated list of [Media Type]s alongside any associated parameters.
+
+It is prefixed by a **length**, which is just a stringified integer (that is, a 32-byte metadata will have "32" as its length). When parsing, you should consume characters until hitting a non-digit, since a valid [Media Type] will never start with a digit. Note that row indices are not included in the metadata length.
+
+There is only one standardized parameter:
+
+  * `l`: this defines the *length* (in _raw_ bytes, not encoded data) of this media type within the binary data
+
+You may include others as you see fit.
+
+If there is more than one [Media Type], the length parameter `l` _must_ be set for all but the last type. If there is only one, it may be omitted entirely.
```

Comment:
```suggestion
If there is more than one [Media Type], the size parameter `s` _must_ be set for all but the last type. If there is only one, it may be omitted entirely.
```